### PR TITLE
PartDesign: Patterns, return directly if occurrences is 1

### PR DESCRIPTION
**Pull request by Jolbas**
In LinearPattern and PolarPattern. Skip all checks if occurrences is 1. Allows Expressions on Length/Angle that evaluates to 0 if Occurrences is 1.

This PR fixes issue #8935

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9111">pipeline status</a> for feature branch [PR_9111](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9111). Pipeline [821143100](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/821143100) was triggered at [80d2190](https://github.com/FreeCAD/FreeCAD/pull/9111/commits/80d219019b57db8c6a05a4e30ba7b13215a52712). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/PartDesign/App/FeatureLinearPattern.cpp [hunk 1]
```diff
@@ -72,12 +72,15 @@ short LinearPattern::mustExecute() const
 
 const std::list<gp_Trsf> LinearPattern::getTransformations(const std::vector<App::DocumentObject*>)
 {
-    double distance = Length.getValue();
-    if (distance < Precision::Confusion())
-        throw Base::ValueError("Pattern length too small");
     int occurrences = Occurrences.getValue();
+    if (occurrences == 1)
```
Comment (0penBrain) [inline src/Mod/PartDesign/App/FeatureLinearPattern.cpp:76]:
```suggestion
    if (occurrences <= 1)

```
Safer :)

File: src/Mod/PartDesign/App/FeaturePolarPattern.cpp [hunk 2]
```diff
@@ -73,22 +73,16 @@ short PolarPattern::mustExecute() const
 
 const std::list<gp_Trsf> PolarPattern::getTransformations(const std::vector<App::DocumentObject*>)
 {
-    double angle = Angle.getValue();
-    double radians = Base::toRadians<double>(angle);
-    if (radians < Precision::Angular())
-        throw Base::ValueError("Pattern angle too small");
     int occurrences = Occurrences.getValue();
+    if (occurrences == 1)
```
Comment (0penBrain) [inline src/Mod/PartDesign/App/FeaturePolarPattern.cpp:77]:
```suggestion
    if (occurrences <= 1)

```

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9111">pipeline status</a> for feature branch [PR_9111](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9111). Pipeline [822027251](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/822027251) was triggered at [13d28f5](https://github.com/FreeCAD/FreeCAD/pull/9111/commits/13d28f5b7fc5ff95cbb5a25beff5bb9e0ff03b95). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/PartDesign/App/FeatureLinearPattern.cpp [hunk 3]
```diff
@@ -72,12 +72,16 @@ short LinearPattern::mustExecute() const
...
 {
-    double distance = Length.getValue();
-    if (distance < Precision::Confusion())
-        throw Base::ValueError("Pattern length too small");
     int occurrences = Occurrences.getValue();
+    if (occurrences <= 1)
+
+        return {gp_Trsf()};
     if (occurrences < 1)
         throw Base::ValueError("At least one occurrence required");
```
Comment (0penBrain) [inline src/Mod/PartDesign/App/FeatureLinearPattern.cpp:80]:
```suggestion
    if (occurrences < 1)
        throw Base::ValueError("At least one occurrence required");

    if (occurrences == 1)
        return {gp_Trsf()};
```
Sorry @Jolbas I was too fast at reviewing and mixed with some other stuffs. :confused: 
I think it's better as suggested above. Sorry again.

File: src/Mod/PartDesign/App/FeaturePolarPattern.cpp [hunk 4]
```diff
@@ -73,22 +73,17 @@ short PolarPattern::mustExecute() const
...
-    double angle = Angle.getValue();
-    double radians = Base::toRadians<double>(angle);
-    if (radians < Precision::Angular())
-        throw Base::ValueError("Pattern angle too small");
     int occurrences = Occurrences.getValue();
+    if (occurrences <= 1)
+
+        return {gp_Trsf()};
     if (occurrences < 1)
         throw Base::ValueError("At least one occurrence required");
```
Comment (0penBrain) [inline src/Mod/PartDesign/App/FeaturePolarPattern.cpp:81]:
```suggestion
    if (occurrences < 1)
        throw Base::ValueError("At least one occurrence required");
        
    if (occurrences <= 1)
        return {gp_Trsf()};
```

File: src/Mod/PartDesign/App/FeatureLinearPattern.cpp (hunk 3 above)
Comment (Jolbas) [reply src/Mod/PartDesign/App/FeatureLinearPattern.cpp:80]: Yes. No problem. I should have reviewed your suggestion but did just accept it.

File: src/Mod/PartDesign/App/FeaturePolarPattern.cpp (hunk 4 above)
Comment (Jolbas) [reply src/Mod/PartDesign/App/FeaturePolarPattern.cpp]: done

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9111">pipeline status</a> for feature branch [PR_9111](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9111). Pipeline [822196346](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/822196346) was triggered at [fe02ce5](https://github.com/FreeCAD/FreeCAD/pull/9111/commits/fe02ce5162eeff6b257402cb1bddd914254b4008). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by 0penBrain** [comment]
OK, I tested and seems to work well without regression. Merging.

**Comment by Jolbas** [comment]
Thanks

**Pull request merged**